src/graph_impl/stable_graph/serialization.rs

```diff
@@ -197,30 +196,47 @@ where
         E2: Error,
     {
         let ty = PhantomData::<Ty>::from_deserialized(input.edge_property)?;
-        let mut nodes = input.nodes;
         let node_holes = input.node_holes;
         let edges = input.edges;
         if edges.len() >= <Ix as IndexType>::max().index() {
             Err(invalid_length_err::<Ix, _>("edge", edges.len()))?
         }
+        
+        let mut nodes = Vec::with_capacity(input.nodes.len() + node_holes.len());
+
+        let mut compact_nodes = input.nodes.into_iter();
+        let mut node_pos = 0;
+        for hole_pos in node_holes.iter() {
+            nodes.extend(compact_nodes.by_ref().take(hole_pos.index() - node_pos));
```

Comment:
`hole_pos` is arbitrary untrusted input and deserialization's role is to validate it; here `hole_pos - node_pos` can panic in debug builds and wrap around in release builds - shouldn't it be an error at deserialization if the hole position is invalid?

Reply:
Yep. fixed.